lib/dataloader.ex

```diff
@@ -139,6 +139,11 @@ defmodule Dataloader do
     Enum.any?(loader.sources, fn {_name, source} -> Source.pending_batches?(source) end)
   end
 
+  @spec has_source?(t, atom()) :: boolean
+  def has_source?(%{sources: sources}, source) do
+    Map.has_key?(sources, source)
+  end
+
   defp get_source(loader, source_name) do
```

Comment:
Maybe instead we should just make `get_source` not private here? Then you could still do `if !Dataloader.get_source` etc.

Reply:
The reasoning behind using a separate function was that `get_source` will currently raise, and it feels a bit unfriendly to make people wrap their checks in a catch. That said, is there any reason that it needs to raise?